compiler/rustc_lint/src/lints.rs

```diff
@@ -3163,3 +3163,98 @@ pub(crate) struct ReservedMultihash {
     #[suggestion(code = " ", applicability = "machine-applicable")]
     pub suggestion: Span,
 }
+
+pub(crate) struct MismatchedElidedLifetimeStyles {
+    pub lifetime_name: String,
+    pub inputs: Vec<Span>,
+    pub outputs: Vec<Span>,
+
+    pub suggestions: Vec<MismatchedElidedLifetimeStylesSuggestion>,
+}
+
+impl<'a, G: EmissionGuarantee> LintDiagnostic<'a, G> for MismatchedElidedLifetimeStyles {
+    fn decorate_lint<'b>(self, diag: &'b mut Diag<'a, G>) {
+        diag.primary_message(fluent::lint_mismatched_elided_lifetime_styles);
+
+        // TODO: how avoid calling it lifetime_name2?
+        diag.arg("lifetime_name2", self.lifetime_name);
```

Comment:
This is called `lifetime_name2` because the nested `MismatchedElidedLifetimeStylesSuggestion::Named` *also* declares a fluent variable called `lifetime_name` and they clobber each other. I'm pretty sure this is me doing it wrong, but I have not yet found the right way.

Reply:
The commit [Move eager translation to a method on Diag](https://github.com/rust-lang/rust/pull/138677/commits/152e0564d7f191a81681d717e41e79d0575bfd37) allows performing eager translation like subdiagnostics, so the name can be duplicated.